v-next/hardhat-keystore/src/internal/tasks/delete.ts

```diff
@@ -0,0 +1,53 @@
+import type { KeystoreLoader } from "../types.js";
+import type { UserInteractions } from "../ui/user-interactions.js";
+import type { HardhatRuntimeEnvironment } from "@ignored/hardhat-vnext/types/hre";
+import type { NewTaskActionFunction } from "@ignored/hardhat-vnext/types/tasks";
+
+import { setupDirectInterruptionsAndKeystoreLoader } from "../utils/setup-direct-interruptions-and-keystore-loader.js";
+
+interface TaskDeleteArguments {
+  key: string;
+  force: boolean;
+}
+
+const taskDelete: NewTaskActionFunction<TaskDeleteArguments> = async (
+  setArgs,
+  hre: HardhatRuntimeEnvironment,
+): Promise<void> => {
+  const { keystoreLoader, interruptions } =
+    await setupDirectInterruptionsAndKeystoreLoader(hre);
+
+  await remove(setArgs, keystoreLoader, interruptions);
+};
+
+export const remove = async (
+  { key, force }: TaskDeleteArguments,
+  keystoreLoader: KeystoreLoader,
+  interruptions: UserInteractions,
+): Promise<void> => {
+  if (!(await keystoreLoader.isKeystoreUninitialized())) {
+    process.exitCode = 1;
+    return interruptions.displayNoKeystoreSetErrorMessage();
```

Comment:
I am not a fan of this pattern.

We have to add opt outs in the tests to deal with `process.exit`. I think we need framework level support in tasks for dealing with failed tasks.

Reply:
Can you expand on this?

Reply:
It forces the plugin author to put reversing `process.exitCode = undefined` into tests where the task finishes in a state where task failure should be reported.

I can imagine three approachs:
* live with plugin authors having do `process.exitCode` twidling in their tests - though I think we should think through how this composes for `./scripts`.
* enhance the return results of task to allow passing along exitCode information, the test can then check for this while the Hardhat cli sets the `process.exitCode` - this complicates the return result for tasks called from scripts
* enhance the context that the tasks run in to provide a side channel for this information, so

```ts
context.setAsFailedTask()
return myTaskResult;
```

I don't think we have to solve this here - but I think writing a task properly involves specifying which return results should be interpreted as a fail for the cli task.